### PR TITLE
[WIP] Gitlab support

### PR DESCRIPTION
**Pull request by cmckni3**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -15,8 +15,13 @@ GITHUB_TOKEN={fill me in}
 # GOOGLE_CLIENT_SECRET={optional, fill me in}
 # GOOGLE_DOMAIN={optional, set to @company.com to limit login only for people at Company}
 
+# GITLAB_KEY={fill me in}
```
Comment (cmckni3) [inline .env.example]: This needs to be documented in the README

Comment (cmckni3) [inline .env.example]: I can work on adding this on Friday.

**Comment by cmckni3** [comment]
Not sure why `vendor/cache/omniauth-gitlab-1.0.0.gem` was added to the repo but I can remove it.

**Comment by cmckni3** [comment]
I think `config/initializers/octokit.rb` should be tweaked so a GitHub token and GitHub auth are optional. Probably add a check to ensure there are credentials for a supported omniauth provider.

**Comment by sandlerr** [comment]
Yep, we can change the code to allow no Github connection. I think we should have guards around some of the pages that rely on octokit unless Gitlab is somehow compatible with it. For example the tabs under a deploy other than 'Output' seem to rely on github-specific functionality.

We should probably have a `gitlab_project` fixture and run some of our specs against it too.

**Comment by cmckni3** [comment]
I agree. Definitely have some work to do on the GitLab integration side.

**Comment by povilasb** [comment]
@cmckni3 what's the status of this feature?

**Comment by cmckni3** [comment]
@povilasb Still needs the actual integration o GitLab itself.

This only adds the authentication portion. Need to use the GitLab API to display a list of available projects.

**Comment by cmckni3** [comment]
Give me a few minutes to rebase my branch.

**Comment by cmckni3** [comment]
Rebased and fixed a bunch of issues but still need to integrate with GitLab

**Comment by markkuit** [comment]
What's the status of this?

**Comment by cmckni3** [comment]
@markkuit Haven't had time to work on the GitLab integration portion.

You're welcome to fork my fork and contribute.

**Comment by bradjones1** [comment]
How is this different from https://github.com/zendesk/samson/pull/1028 or https://github.com/zendesk/samson/pull/735?

**Comment by cmckni3** [comment]
@bradjones1 I abandoned this last year and never got around to it.

Looks like the functionality exists. PR Closed

**Pull request closed without merging**